transformation_test/transformation_test.go

```diff
@@ -0,0 +1,216 @@
+package transformation_test
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+	"github.com/goccy/go-yaml"
+	"github.com/google/go-cmp/cmp"
+	"golang.org/x/sync/errgroup"
+	"gotest.tools/v3/golden"
+)
+
+const (
+	flbMainConf          = "fluent_bit_main.conf"
+	flbParserConf        = "fluent_bit_parser.conf"
+	transformationInput  = "input.log"
+	transformationOutput = "output_fluentbit.yaml"
+	flbTag               = "transformation_test"
+)
+
+var (
+	flbPath = flag.String("flb", os.Getenv("FLB"), "Fluent-bit path")
+)
+
+type transformationTest []loggingProcessor
+type loggingProcessor struct {
+	confgenerator.LoggingProcessor
+}
+
+func (l *loggingProcessor) UnmarshalYAML(ctx context.Context, unmarshal func(interface{}) error) error {
+	return confgenerator.LoggingProcessorTypes.UnmarshalComponentYaml(ctx, &l.LoggingProcessor, unmarshal)
+}
+
+func TestTransformationTests(t *testing.T) {
+	ctx := context.Background()
+	if len(*flbPath) == 0 {
+		t.Skip("--flb not supplied")
+	}
+
+	allTests, err := os.ReadDir("testdata")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, dir := range allTests {
+		dir := dir
+		if !dir.IsDir() {
+			continue
+		}
+		t.Run(dir.Name(), func(t *testing.T) {
+			t.Parallel()
+			testStartTime := time.Now()
+			// Unmarshal transformation_config.yaml
+			var transformationConfig transformationTest
+			transformationConfig, err = readTransformationConfig(dir.Name())
+			if err != nil {
+				t.Fatal("failed to unmarshal config:", err)
+			}
+
+			// Generate config files
+			var genFiles map[string]string
+			genFiles, err = generateFluentBitConfigs(ctx, transformationConfig, dir.Name())
+
+			// Write config files in temp directory
+			tempPath := t.TempDir()
+			for k, v := range genFiles {
+				err := confgenerator.WriteConfigFile([]byte(v), filepath.Join(tempPath, k))
+
+				if err != nil {
+					t.Fatal(err)
+				}
+				t.Logf("generated file %q\n%s", k, v)
+			}
+
+			// Start Fluent-bit
+			cmd := exec.Command(
+				fmt.Sprintf("%s/fluent-bit", *flbPath),
```

Comment:
Why does `flbPath` point to the executable's parent folder rather than the executable itself?